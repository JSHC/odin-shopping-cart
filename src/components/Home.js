import ShopItem from "./ShopItem";
import '../styles/Home.css';

const Home = (props) => {
    const shopItems = [
        {id: 1, name: 'Item1', price: 10, image: 'https://images.unsplash.com/photo-1608256246200-53e635b5b65f'},
        {id: 2, name: 'Item2', price: 20, image: ''}
    ]

    const onAddToCart = (item) => {
        if (props.onAddToCart) {
            props.onAddToCart(item);
        }
    }
    return (
        <div className="home-container">
            <h1>Home</h1>
            <div className="shop-items-container">
                {shopItems.map(item => {
                    return <ShopItem {...item} onAddToCart={onAddToCart} key={item.id}/>
                })}
            </div>
        </div>
    );
};

export default Home;
