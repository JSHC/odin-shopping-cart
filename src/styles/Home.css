.shop-items-container {
    display: grid;
    grid-auto-flow: column;
    gap: 20px;
    grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
}