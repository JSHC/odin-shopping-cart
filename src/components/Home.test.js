import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from './Home';
import { MemoryRouter } from 'react-router-dom';

describe('Home component', () => {
    it('renders', () => {
        const { container } = render(
        <MemoryRouter>
        <Home />
        </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    }) 
})