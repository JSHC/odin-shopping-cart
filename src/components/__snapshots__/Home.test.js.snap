// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders 1`] = `
<div>
  <div>
    <h1>
      Home
    </h1>
    <div
      class="shop-items-container"
    >
      <div
        class="shop-item"
      >
        <span
          class="shop-item-name"
        >
          Item1
        </span>
        <span
          class="shop-item-price"
        >
          10
        </span>
        <div
          class="shop-item-controls"
        >
          <label
            for="shop-item-quantity"
          >
            Quantity
          </label>
          <input
            id="shop-item-quantity"
            name="shop-item-quantity"
            type="text"
            value="0"
          />
          <button>
            Inc
          </button>
          <button>
            Dec
          </button>
          <button>
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
